lib/ansible/config/manager.py

```diff
@@ -287,7 +287,7 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                         for ini_entry in defs[config]['ini']:
                             value = get_ini_config_value(self._parser, ini_entry)
                             origin = cfile
-                            if 'deprecated' in ini_entry:
+                            if 'deprecated' in ini_entry and value:
```

Comment:
if a value is 'False' it should not be deprecated either, full fix here:
https://github.com/ansible/ansible/pull/29542/files#diff-0c625695bed181c8ccc287e78a55d40eR299